### ISSUE TITLE
Http Reporter always calling http://localhost

### ISSUE THREAD

**Issue by andrei-dascalu** [openzipkin/zipkin-php#112]
Hello,

While trying to use zipkin-php in a docker-compose context, I found that traces sent to a zipkin server running inside compose network from a php script also running in compose would not be picked up, whereas the same script sent from local machine to the same zipkin (but running with exposed port 9411) would work ok.

I found that:

- as per zipkin-php-example code, simply creating the endpoint with `$endpoint = Endpoint::create($endpointName, $ipv4, null, $port);` doesn't have an impact on where the HttpReporter makes api calls (I would be expecting it calls a server found at $ipv4 ip)
- passing an options array specifying the url to the Http reporter constructor works (`endpoint_url` key)

How `create_tracing` method looks now:

`
    $endpoint = Endpoint::create($endpointName, $ipv4, null, $port);
    $http = \Zipkin\Reporters\Http\CurlFactory::create();
    $reporter = new Zipkin\Reporters\Http($http, [
        'endpoint_url' => sprintf('http://%s:%s/api/v2/spans', $ipv4, $port)
    ]);
    $sampler = BinarySampler::createAsAlwaysSample();
    $tracing = TracingBuilder::create()
        ->havingLocalEndpoint($endpoint)
        ->havingSampler($sampler)
        ->havingReporter($reporter)
        ->build();
`

I'm not entirely sure whether my understanding is completely off (eg: is the $ipv4 where api calls should be made?) or whether this is a bug and what would be a way to handle it.

I believe at the very least the docs should reflect this.

Cheers

**Comment by jcchavezs** [comment]
Thanks for such a detailed issue. Could please also link the repository you
uploaded your code? Thanks!

lør. 18. mai 2019, 07:11 skrev Andrei Dascalu <notifications@github.com>:

**Comment by andrei-dascalu** [comment]
Hi,

The repository is at https://github.com/andrei-dascalu/zipking-php-testing - with 2 branches (`master` with the original code based on zipkin-php-example, having put everything in one file and removed calls to a backend and `fix` showing the fix mention above)

Cheers

**Comment by jcchavezs** [comment]
@andrei-dascalu sorry I did not come back to you before. Is this still an issue? if that is the case I will try your repo this weekend.

**Comment by andrei-dascalu** [comment]
@jcchavezs also sorry for the delay, but yes it's still an issue

**Comment by jcchavezs** [comment]
Hi @andrei-dascalu 

I checked this issue in dept. I think there is a confusion about the usage of the endpointName and the ipv4 in the `create_tracing` function. Sorry for that, as you correctly pointed out, `endpointName` and the `ipv4` has nothing to do with the reporter URL. I added comments for this and also added the possibility to pass a custom reporter URL int he example through an env var. Check https://github.com/openzipkin/zipkin-php/issues/112. Feel free to ping me directly in gitter if further questions: https://gitter.im/openzipkin/zipkin
